tests/Schema/TableTest.php

```diff
@@ -50,6 +51,26 @@ public function testColumns(): void
         self::assertCount(2, $table->getColumns());
     }
 
+    /** @dataProvider getNormalizesAssetNames */
+    public function testColumnsOrder(string $assetName): void
+    {
+        $type      = Type::getType(Types::INTEGER);
+        $columns   = [];
+        $columns[] = new Column('bar', $type);
+        $columns[] = new Column('BAZ', $type);
+        $columns[] = new Column($assetName, $type);
+        $table     = new Table('foo', $columns, [], []);
+
+        $table->setPrimaryKey([$assetName]);
+        $table->addForeignKeyConstraint('bar', ['BAZ'], ['baz']);
+
+        $orderedColumns = array_values($table->getColumns());
+
+        self::assertSame($columns[2], $orderedColumns[0]);
+        self::assertSame($columns[1], $orderedColumns[1]);
+        self::assertSame($columns[0], $orderedColumns[2]);
```

Comment:
I'm sorry, these assertions are impossible to comprehend. What are you trying to achieve?

Reply:
My goal was to fix the inconsistent behavior seen in #4158
But I guess that will become moot if the (bugged) ordering is removed altogether 🤔